flag_test.go

```diff
@@ -2284,9 +2284,37 @@ func TestTimestampFlagApply(t *testing.T) {
 	assert.Equal(t, expectedResult, set.Lookup("time").Value.(flag.Getter).Get())
 }
 
+func TestTimestampFlagApply_MultipleFormats(t *testing.T) {
+	fl := TimestampFlag{
+		Name:    "time",
+		Aliases: []string{"t"},
+		Config: TimestampConfig{
+			Timezone: time.Local,
+			AvailableLayouts: []string{
+				time.DateTime,
+				time.TimeOnly,
+				time.Kitchen,
+			},
+		},
+	}
+	set := flag.NewFlagSet("test", 0)
+	_ = fl.Apply(set)
+
+	now := time.Now()
+	for timeStr, expectedRes := range map[string]time.Time{
+		now.Format(time.DateTime): now.Truncate(time.Second),
+		now.Format(time.TimeOnly): now.Truncate(time.Second),
+		now.Format(time.Kitchen):  now.Truncate(time.Minute),
+	} {
+		err := set.Parse([]string{"--time", timeStr})
+		assert.NoError(t, err)
+		assert.Equal(t, expectedRes, set.Lookup("time").Value.(flag.Getter).Get())
+	}
+}
```

Comment:
Because the layout is now changed from `Layout string` to `AvailableLayouts []string`, we need to test more edge cases such as

1. A `nil` `AvailableLayouts`
2. An empty `AvailableLayouts`
3. Combination of invalid and valid layouts
4. All invalid layouts
5. etc.

You could use [Table-driven tests style](https://go.dev/wiki/TableDrivenTests).

Reply:
Thanks for the feedback. New tests have been added